File: frontend/src/types/order.ts
export type OrderStatus = "PENDING" | "PROCESSING" | "COMPLETED" | "CANCELLED";

export interface OrderItem {
  id: number;
  product_id: number;
  product_name: string;
  quantity: number;
  unit_price: number;
  subtotal: number;
}

export interface Order {
  id: number;
  customer_id: number;
  customer_name: string;
  status: OrderStatus;
  total: number;
  created_at: string;
  updated_at: string;
  items: OrderItem[];
}

export type OrderItemCreate = {
  product_id: number;
  quantity: number;
};

export type OrderCreate = {
  customer_id: number;
  items: OrderItemCreate[];
};

File: backend/app/schemas/order.py
from datetime import datetime

from pydantic import BaseModel, Field

from app.models.order import OrderStatus


class OrderItemCreate(BaseModel):
    product_id: int
    quantity: int = Field(..., ge=1)


class OrderCreate(BaseModel):
    customer_id: int
    items: list[OrderItemCreate] = Field(..., min_length=1)


class OrderStatusUpdate(BaseModel):
    status: OrderStatus


class OrderItemResponse(BaseModel):
    id: int
    product_id: int
    product_name: str
    quantity: int
    unit_price: float
    subtotal: float


class OrderResponse(BaseModel):
    id: int
    customer_id: int
    customer_name: str
    status: OrderStatus
    total: float
    created_at: datetime
    updated_at: datetime
    items: list[OrderItemResponse]
